libnmstate/netapplier.py

```diff
@@ -118,13 +118,15 @@ def _apply_ifaces_state(
     desired_state, verify_change, commit, rollback_timeout
 ):
     current_state = state.State(netinfo.show())
-
-    desired_state.sanitize_ethernet(current_state)
-    desired_state.sanitize_dynamic_ip()
-    desired_state.merge_routes(current_state)
-    desired_state.merge_dns(current_state)
-    metadata.generate_ifaces_metadata(desired_state, current_state)
-
+    vlan_ifaces_state = _prepare_vlan_ifaces_for_mtu_change(
+        desired_state, current_state
+    )
+    desired_state.merge_interfaces(
+        state.State({schema.Interface.KEY: vlan_ifaces_state})
+    )
+    desired_state = _prepare_desired_state_for_apply(
```

Comment:
Extracting to a function is always good, but now it is not related to this change, therefore belonging to a refactor commit.
But, I also want to state that the name is not helping here, it tells little of what is done (or what it represents). I suggest you extract this to a separate PR for discussion.

Reply:
I will do that in a separate PR. Thanks!